shell/renderer/renderer_client_base.cc

```diff
@@ -361,11 +361,23 @@ bool RendererClientBase::IsPluginHandledExternally(
 
 bool RendererClientBase::IsOriginIsolatedPepperPlugin(
     const base::FilePath& plugin_path) {
-#if BUILDFLAG(ENABLE_PDF_VIEWER)
-  return plugin_path.value() == kPdfPluginPath;
-#else
-  return false;
-#endif
+  // Taken from Chrome's own implementation of this same method:
+  // https://source.chromium.org/chromium/chromium/src/+/37893bf803b256be100b1bd1805abcd069c33140:chrome/renderer/chrome_content_renderer_client.cc;l=1341-1363
+  //
+  // NOTE: the logic for NaCl is excluded here as we don't currently support it.
+
+  // Hosting plugins in-process is inherently incompatible with attempting to
+  // process-isolate plugins from different origins.
+  auto* cmdline = base::CommandLine::ForCurrentProcess();
+  if (cmdline->HasSwitch(::switches::kPpapiInProcess)) {
+    // The kPpapiInProcess switch should only be used by tests.
+    CHECK_NE(kPdfPluginPath, plugin_path.value());
+
+    return false;
+  }
```

Comment:
Hm, given (a) we don't really support `--ppapi-in-process` ("The kPpapiInProcess switch should only be used by tests") and (b) this `CHECK` will trigger if you pass that flag with this code here, since we always load the pdf plugin, I think we can actually exclude this whole thing and just have this function `return true` 🤔

Reply:
Oh my bad, for some reason I assumed that test behavior would also apply to our own tests, and I did not consider that the PDF plugin always loads therefore always triggering this `CHECK` 🤕

Thank you for the insight 🙇